nni/retiarii/trainer/pytorch/base.py

```diff
@@ -188,57 +213,64 @@ def __init__(self, multi_model, kwargs=[]):
-                        
-                optimizer = getattr(torch.optim, optimizer_cls)(one_model_params, **(optimizer_kwargs or {}))
+
+                optimizer = getattr(torch.optim, optimizer_cls)(
+                    one_model_params, **(optimizer_kwargs or {}))
                 self._optimizers.append(optimizer)
 
     def fit(self) -> None:
         torch.autograd.set_detect_anomaly(True)
-        max_epochs = max([x['trainer_kwargs']['max_epochs'] for x in self.kwargs['model_kwargs']])
+        max_epochs = max([x['trainer_kwargs']['max_epochs']
+                          for x in self.kwargs['model_kwargs']])
         for _ in range(max_epochs):
             self._train()
+        nni.report_final_result(self._validate())
 
     def _train(self):
-        for batch_idx, multi_model_batch in enumerate(zip(*self._dataloaders)):
+        for batch_idx, multi_model_batch in enumerate(zip(*self._train_dataloaders)):
             for opt in self._optimizers:
                 opt.zero_grad()
             xs = []
             ys = []
             for idx, batch in enumerate(multi_model_batch):
-                x, y = self.training_step_before_model(batch, batch_idx, f'cuda:{idx}')
+                x, y = self.training_step_before_model(
+                    batch, batch_idx, f'cuda:{idx}')
                 xs.append(x)
                 ys.append(y)
-            
+
             y_hats = self.multi_model(*xs)
             if len(ys) != len(xs):
                 raise ValueError('len(ys) should be equal to len(xs)')
             losses = []
             report_loss = {}
             for output_idx, yhat in enumerate(y_hats):
                 if len(ys) == len(y_hats):
-                    loss = self.training_step_after_model(xs[output_idx], ys[output_idx], yhat)
+                    loss = self.training_step_after_model(
+                        xs[output_idx], ys[output_idx], yhat)
                 elif len(ys) == 1:
-                    loss = self.training_step_after_model(xs[0], ys[0].to(yhat.get_device()), yhat)
+                    loss = self.training_step_after_model(
+                        xs[0], ys[0].to(yhat.get_device()), yhat)
                 else:
-                    raise ValueError('len(ys) should be either 1 or len(y_hats)')
+                    raise ValueError(
+                        'len(ys) should be either 1 or len(y_hats)')
                 losses.append(loss.to("cuda:0"))
-                report_loss[self.kwargs['model_kwargs'][output_idx]['model_id']] = loss.item()
+                report_loss[self.kwargs['model_kwargs']
+                            [output_idx]['model_id']] = loss.item()
             summed_loss = sum(losses)
             summed_loss.backward()
             for opt in self._optimizers:
                 opt.step()
-            if batch_idx % 50 == 0:
-                nni.report_intermediate_result(report_loss)
+            # if batch_idx % 50 == 0:
```

Comment:
Why comment this?

Reply:
It was for debug. training_loss is not reported. Removed.